### DISCUSSION TITLE
[Enhancement] [table-column] table的show-summary和summary-method

### DISCUSSION THREAD

**Discussion by jinceon** [element-plus/element-plus#17077] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `table-column`

## Description

我需要展示表尾的合计行。
但是
1、有些列是数字形式，但是我不希望它参与合计。
2、有些列需要求平均值，等等。
当然我知道你们在el-table上提供了summary-method，可以自定义实现合并行逻辑。
但是这个summary-method太大了。我需要关注太多我本来可以不关注的列。

我希望可以在el-table-column里增加summary-method。
整个逻辑可以变成类似：
element-plus/packages/components/table/src/table-footer/index.ts

```vue
let sums = []
    if (summaryMethod) {
      sums = summaryMethod({
        columns,
        data,
      })
    } else {
      columns.forEach((column, index) => {
        if (index === 0) {
          sums[index] = sumText
          return
        }
        // 这里插入类似的判断
        if (column.summaryMethod){
            sums[index] = column.summaryMethod( {columns, data} );
            return
        }
```

### Additional comments

_(empty)_

![image](https://github.com/element-plus/element-plus/assets/1641265/ac1441c8-58ba-42b2-9326-3f866a0b50bf)

**Comment by jinceon** [comment]
summaryMethod最好能提供常见的2种：sum, average, ignore(忽略这列，显示空白)

**Reply by xxholly32** [reply]
感觉是一种偏见，相当于是把软件复杂度从业务代码转移到了组件库

**Reply by jinceon** [reply]
你这个才是偏见吧。
sum和average是同一个维度的东西，不是吗？
所以把默认的sum改成由用户选择是sum还是average还是ignore，是一个局部的设计调整，完全不违背整个组件的设计理念

**Reply by xxholly32** [reply]
summary 是摘要总结的意思，并不是仅代表合计或者平均值，单元格也有各种形式，并不是只有数字，如果官方这么做了，那其他会有提案让excel的所有公式其实都做进去，所以我说会增加软件的复杂度

如果觉得业务实现比较麻烦，可以在业务中写个通用的方法来实现 summary-method 里面的生成过程

![image](https://github.com/element-plus/element-plus/assets/6063358/71423000-baa5-4b24-bbf7-d9fee69c5410)

**Reply by jinceon** [reply]
>如果觉得业务实现比较麻烦，可以在业务中写个通用的方法来实现 summary-method 里面的生成过程

我现在就是这样做的
https://gitee.com/jinceon/el-table-summary-method/blob/master/src/components/TheWelcome.vue

**Comment by jinceon** [comment]
> 如果官方这么做了，那其他会有提案让excel的所有公式其实都做进去，所以我说会增加软件的复杂度

这个想多了，现在缺的是一个el-table-column 上的summary-method钩子函数。这个函数只能靠组件内部实现。至于其他的公式，只要有了这个钩子，其他函数开发者自己想怎么传就怎么传

summary-method函数同时放在el-table和el-table-column上，和rules同时放在el-form和el-form-item上，有什么区别吗？
